native/dggsvp3.go

```diff
@@ -260,8 +260,8 @@ func (impl Implementation) Dggsvp3(jobU, jobV, jobQ lapack.GSVDJob, m, p, n int,
 		}
 
 		// Clean up A.
-		for i := k + 1; i < k+l; i++ {
-			r := a[i*lda+n-l : i*lda+i-k+n-l]
+		for i := k + 1; i < m; i++ {
+			r := a[i*lda+n-l : i*lda+min(i-k+n-l, n)]
```

Comment:
Just a nit: if it were written as `n-l+i-k`, then (at least for me) it's easier to see that `n-l` is a fixed offset and `i-k` a variable index, and what the loop does.